src/main/java/mate/academy/springboot/datajpa/persistance/ProductRepository.java

```diff
@@ -0,0 +1,15 @@
+package mate.academy.springboot.datajpa.persistance;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ProductRepository extends JpaRepository<Product, Long> {
+    List<Product> findAllByPriceBetween(BigDecimal min, BigDecimal max);
+
+    List<Product> findAllByCategoryIn(List<Category> categories);
```

Comment:
you can try using something like `List<Product> findAllByCategoryIdIn(List<Long> categoryIds);` repository method also has to work and simplify your solution (or find by category names)